### PR TITLE
feat(experience): add user experience tab tracking

### PR DESCRIPTION
**Pull request by overbalance**
## Summary
Introduces **EmbraceExperienceManager** to track user journeys across browser tabs by assigning persistent experience IDs that group related sessions.

## Key Features
- **Experience ID**: Persistent identifier shared across related tabs
- **Tab Detection**: Identifies how tabs are opened (link, manual, window.open, reload, back/forward)  
- **Cross-Tab Inheritance**: Child tabs inherit parent's experience ID via localStorage
- **Referrer Tracking**: Captures referrer type, domain (external), or path (same-origin)

## New Attributes
```typescript
emb.experience_id      // Groups related sessions
emb.tab_open_method    // How tab was opened
emb.referrer_type      // same_origin/external/none
emb.referrer_path      // For same-origin navigation
emb.referrer_domain    // For external referrers
emb.previous_tab_id    // Parent tab ID if inherited
```

## API Additions
```typescript
sessionManager.getExperienceId(): string | null
sessionManager.getPreviousTabId(): string | null
```

## Technical Details
- Dual-key storage system for accurate parent-child matching
- Auto-cleanup of data older than 24 hours
- Comprehensive test coverage (779 lines, all passing)
- Performance optimized with pre-computed referrer info

## Testing
✅ All tests passing - covers detection, inheritance, storage quota handling, and edge cases

### CODE REVIEW

File: src/resources/experienceId.ts
[the remark this answers is not in the document]
Comment (overbalance) [reply src/resources/experienceId.ts:1]: I'm going to refactor and we can review together.

File: src/resources/experienceId.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import { generateUUID } from '../utils/index.js';
+
+const EXPERIENCE_ID_KEY = 'embrace_experience_id';
```
[the remark this answers is not in the document]
Comment (overbalance) [reply src/resources/experienceId.ts]: I'm going to refactor and we can review together.

File: src/managers/EmbraceSpanSessionManager/EmbraceSpanSessionManager.ts [hunk 2]
```diff
@@ -208,6 +212,7 @@ export class EmbraceSpanSessionManager implements SpanSessionManagerInternal {
 
     this._sessionSpan.setAttributes({
       ...this._getPermanentAttributes(),
+      ...this._experienceManager.getExperienceAttributes(),
```
Comment (overbalance) [inline src/managers/EmbraceSpanSessionManager/EmbraceSpanSessionManager.ts:215]: Calling this twice: once at startSession and again at endSession. Only because you may want to read the attributes in the middle of a session. The second call is needed to update the lastActivity update.

File: src/managers/EmbraceSpanSessionManager/EmbraceSpanSessionManager.ts [hunk 3]
```diff
@@ -257,6 +266,7 @@ export class EmbraceSpanSessionManager implements SpanSessionManagerInternal {
 
     const attributes: Attributes = {
       ...this._getPermanentAttributes(),
+      ...this._experienceManager.getExperienceAttributes(),
```
Comment (overbalance) [inline src/managers/EmbraceSpanSessionManager/EmbraceSpanSessionManager.ts:269]: Here is the second call.

File: src/managers/EmbraceSpanSessionManager/types.ts [hunk 4]
```diff
@@ -3,13 +3,15 @@ import type { PerformanceManager } from '../../utils/index.js';
...
 import type { LimitManagerInternal } from '../EmbraceLimitManager/index.js';
+import type { EmbraceExperienceManager } from '../EmbraceExperienceManager/index.js';
 
 export interface EmbraceSpanSessionManagerArgs {
   diag?: DiagLogger;
   perf?: PerformanceManager;
   visibilityDoc?: VisibilityStateDocument;
   storage?: Storage;
   limitManager: LimitManagerInternal;
+  experienceManager?: EmbraceExperienceManager;
```
Comment (overbalance) [inline src/managers/EmbraceSpanSessionManager/types.ts:14]: Set as optional because I only foresee this used in testing.

File: src/managers/EmbraceExperienceManager/EmbraceExperienceManager.ts [hunk 5]
```diff
@@ -0,0 +1,576 @@
...
+    try {
+      // Store with tab-specific key
+      this._storage.setItem(
+        `${INHERITANCE_KEY_PREFIX}${this._appInstanceId}`,
+        experienceDataStr
+      );
+
+      // Store with referrer-based key for accurate child matching
+      const referrerKey = EmbraceExperienceManager._createReferrerKey(
+        window.location.href
```
Comment (joaquin-diaz) [inline src/managers/EmbraceExperienceManager/EmbraceExperienceManager.ts:338]: Should we also provide a location object via constructor as we do in other places?

File: src/managers/EmbraceExperienceManager/EmbraceExperienceManager.ts [hunk 6]
```diff
@@ -0,0 +1,576 @@
...
+      // Ignore URL parsing errors
+    }
+  }
+
+  /**
+   * Update the last activity timestamp
+   * Called on session start/end events
+   */
+  public updateLastActivity(): void {
+    this._experienceData.lastActivityAt = Date.now();
```
Comment (joaquin-diaz) [inline src/managers/EmbraceExperienceManager/EmbraceExperienceManager.ts:542]: If lastActivityAt is only based on session, maybe we should rename it to something else. Activity can be confused as user activity like moving the cursor or clicking, also if it only depends on session activity maybe we can use the session lifecycle listeners? we have `addSessionEndedListener` and `addSessionStartedListener`

File: src/managers/EmbraceExperienceManager/EmbraceExperienceManager.ts [hunk 7]
```diff
@@ -0,0 +1,576 @@
...
+    this._storeExperienceData();
+  }
+
+  /**
+   * Get all experience attributes for session spans
+   * Updates last activity timestamp on each call
+   */
+  public getExperienceAttributes(): Record<string, AttributeValue> {
+    // Update last activity timestamp
+    this.updateLastActivity();
```
Comment (joaquin-diaz) [inline src/managers/EmbraceExperienceManager/EmbraceExperienceManager.ts:552]: It feels weird that this gets updated when you get the attributes, this is only done this way because we know we call it on session start / end?

File: src/managers/EmbraceExperienceManager/EmbraceExperienceManager.test.ts [hunk 8]
```diff
@@ -0,0 +1,789 @@
...
+      expect(manager.getTabOpenMethod()).to.equal('manual_new_tab');
+    });
+  });
+
+  describe('inheritance', () => {
+    it('should inherit experience ID from parent tab via window.opener', () => {
+      // First ensure window.opener is null (from beforeEach)
+
+      // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+      expect(window.opener).to.be.null;
```
Comment (joaquin-diaz) [inline src/managers/EmbraceExperienceManager/EmbraceExperienceManager.test.ts:215]: I think you can avoid all the `eslint-disable-next-line @typescript-eslint/no-unused-expressions` by doing `void expect(window.opener).to.be.null;`

File: src/api-sessions/api/SessionAPI/SessionAPI.ts [hunk 9]
```diff
@@ -62,6 +63,10 @@ export class SessionAPI implements SpanSessionManager {
     return this.getSpanSessionManager().getSessionId();
   }
 
+  public getExperienceManager(): EmbraceExperienceManager | null {
+    return this.getSpanSessionManager().getExperienceManager();
```
Comment (joaquin-diaz) [inline src/api-sessions/api/SessionAPI/SessionAPI.ts:67]: I'm wondering if the experience should be it's own global API, given that sessions are current-tab related and experiences is cross tab

**Review by joaquin-diaz — COMMENTED** [review]
A few questions around activity but the rest looks amazing!

File: src/managers/EmbraceSpanSessionManager/EmbraceSpanSessionManager.ts (hunk 2 above)
Comment (overbalance) [reply src/managers/EmbraceSpanSessionManager/EmbraceSpanSessionManager.ts:215]: my mistake. I did this when I decided to report last activity time, but decided to only use it internally to determine which experience. There *are* no updates to the values within a session. I'll remove.

File: src/managers/EmbraceExperienceManager/EmbraceExperienceManager.ts (hunk 7 above)
Comment (overbalance) [reply src/managers/EmbraceExperienceManager/EmbraceExperienceManager.ts:552]: My idea is that updates would come from multiple places in the code depending on what we consider activity. This happens to be the only one so far. I think it looks weird because we aren't tracking other user interactions yet.

File: src/managers/EmbraceExperienceManager/EmbraceExperienceManager.ts (hunk 6 above)
Comment (overbalance) [reply src/managers/EmbraceExperienceManager/EmbraceExperienceManager.ts:540]: My comment missed your point @joaquin-diaz, I was responding to @jpmunz.

I did remove the attribute from the data, but last activity is used to calculate the experience of the next tab. Removing it wouldn't make sense as this is architected currently. It's a combination of referrer and most recent using a best-candidate algo.
